nestkernel/connection_manager.cpp

```diff
@@ -356,6 +356,15 @@ nest::ConnectionManager::connect( NodeCollectionPTR sources,
   const DictionaryDatum& conn_spec,
   const DictionaryDatum& syn_spec )
 {
+  if ( sources->empty() )
+  {
+    throw BadProperty( "Presynaptic nodes cannot be an empty NodeCollection" );
```

Comment:
I think `BadProperty` is not the right exception to raise. `BadProperty` is meant, for, e.g. negative time constants. I would suggest `IllegalConnection` instead.